tracer/src/Datadog.Trace/Ci/CITracerManagerFactory.cs

```diff
@@ -46,7 +48,14 @@ public CITracerManagerFactory(CIVisibilitySettings settings, IDiscoveryService d
             string defaultServiceName,
             IGitMetadataTagsProvider gitMetadataTagsProvider)
         {
-            return new CITracerManager(settings, agentWriter, sampler, scopeManager, statsd, runtimeMetrics, logSubmissionManager, telemetry, discoveryService, dataStreamsManager, defaultServiceName, gitMetadataTagsProvider);
+            if (_useLockedManager)
```

Comment:
It seems like all the complexity around needing a _sometimes_ locked, _sometimes_ unlocked tracer is because you run the test in-memory in `CiRunCommandTests` right? Would it be cleaner to run that as a "normal" integration test (that runs out-of-proc). That way none of this would need to change, no? 🤔